File: frontend/src/types/estudiante.ts
import type { Especialidad } from './especialidad'
import type { Usuario } from './usuario'

export interface Estudiante {
  id: number
  usuario: Usuario
  codigoEstudiante: string
  dni: string
  especialidad: Especialidad
  cicloActual: number
  /** Año de ingreso: define la cohorte */
  anioIngreso: number
  telefono?: string
  direccion?: string
  fechaNacimiento?: string // ISO date "YYYY-MM-DD"
}

File: sistema-academico/src/main/java/com/universidad/sistema_academico/entity/Estudiante.java
package com.universidad.sistema_academico.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

@Data
@Entity
@Table(name = "estudiantes")
public class Estudiante {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false)
    @JoinColumn(name = "usuario_id", unique = true)
    private Usuario usuario;

    @Column(unique = true, nullable = false, length = 15)
    private String codigoEstudiante;

    @Column(unique = true, nullable = false, length = 8)
    private String dni;

    @ManyToOne(optional = false)
    @JoinColumn(name = "especialidad_id")
    private Especialidad especialidad;

    @Column(nullable = false)
    private Integer cicloActual = 1;

    @Column(nullable = false)
    private Integer anioIngreso;

    private String telefono;

    private String direccion;

    private LocalDate fechaNacimiento;
}
